pkg/k8shandler/pod_test.go

```diff
@@ -0,0 +1,88 @@
+package k8shandler
+
+import (
+	"testing"
+
+	"github.com/openshift/cluster-logging-operator/pkg/utils"
+	v1 "k8s.io/api/core/v1"
+)
+
+func TestNodeAllocationLabelsForPod(t *testing.T) {
+
+	// Create pod with nil selectors, we expect a new selectors map will be created
+	// and it will contain only linux allocation selector.
+	podSpec := NewPodSpec(
+		"Foo",
+		[]v1.Container{},
+		[]v1.Volume{},
+		nil,
+	)
+
+	if podSpec.NodeSelector == nil {
+		t.Errorf("Exp. the nodeSelector to contains the linux allocation selector but was %T", podSpec.NodeSelector)
+	}
+	if len(podSpec.NodeSelector) != 1 {
+		t.Errorf("Exp. single nodeSelector but %d were found", len(podSpec.NodeSelector))
+	}
+	if podSpec.NodeSelector[utils.OsNodeLabel] != utils.LinuxValue {
+		t.Errorf("Exp. the nodeSelector to contains %s: %s pair", utils.OsNodeLabel, utils.LinuxValue)
+	}
+
+	// Create pod with some "foo" selector, we expect a new linux box selector will be added
+	// while existing selectors will be left intact.
+	podSpec = NewPodSpec(
+		"Foo",
+		[]v1.Container{},
+		[]v1.Volume{},
+		map[string]string{"foo": "bar"},
+	)
+
+	if podSpec.NodeSelector == nil {
+		t.Errorf("Exp. the nodeSelector to contains the linux allocation selector but was %T", podSpec.NodeSelector)
+	}
+	if len(podSpec.NodeSelector) != 2 {
+		t.Errorf("Exp. single nodeSelector but %d were found", len(podSpec.NodeSelector))
+	}
+	if podSpec.NodeSelector["foo"] != "bar" {
+		t.Errorf("Exp. the nodeSelector to contains %s: %s pair", "foo", "bar")
+	}
+	if podSpec.NodeSelector[utils.OsNodeLabel] != utils.LinuxValue {
+		t.Errorf("Exp. the nodeSelector to contains %s: %s pair", utils.OsNodeLabel, utils.LinuxValue)
+	}
+
+	// Create pod with "linux" selector, we expect will stay unchanged.
+	podSpec = NewPodSpec(
+		"Foo",
+		[]v1.Container{},
+		[]v1.Volume{},
+		map[string]string{utils.OsNodeLabel: utils.LinuxValue},
+	)
+
+	if podSpec.NodeSelector == nil {
+		t.Errorf("Exp. the nodeSelector to contains the linux allocation selector but was %T", podSpec.NodeSelector)
+	}
+	if len(podSpec.NodeSelector) != 1 {
+		t.Errorf("Exp. single nodeSelector but %d were found", len(podSpec.NodeSelector))
+	}
+	if podSpec.NodeSelector[utils.OsNodeLabel] != utils.LinuxValue {
+		t.Errorf("Exp. the nodeSelector to contains %s: %s pair", utils.OsNodeLabel, utils.LinuxValue)
+	}
+
+	// Create pod with some "non-linux" selector, we expect will stay unchanged.
```

Comment:
why do we expect this to be unchanged? shouldn't `EnsureLinuxNodeSelector` always set the OS node label?

Reply:
Fixed.